app/services/projects_overview_service.rb

```diff
@@ -7,6 +7,14 @@ def initialize(team, user, folder, params)
     @current_folder = folder
     @params = params
     @view_state = @team.current_view_state(@user)
+
+    # Update view_mode if changed
+    if @view_state.state.dig('projects', 'view_mode') != @params[:filter] &&
```

Comment:
Why it is here?

Reply:
I think it was here before to store the current view mode. Otherwise, `@view_mode` will return nil every time.

Reply:
view mode shouldn't be saved